cmdstanpy/stanfit.py

```diff
@@ -229,10 +192,35 @@ def chain_ids(self) -> List[int]:
         """Chain ids."""
         return self._chain_ids
 
-    @property
-    def cmds(self) -> List[List[str]]:
-        """List of call(s) to CmdStan, one call per-chain."""
-        return self._cmds
+    def cmd(self, idx: int) -> List[str]:
+        """
+        Assemble CmdStan invocation.
+        When running parallel chains from single process (2.28 and up),
+        specify CmdStan arg `num_chains` and leave chain idx off CSV files.
+        """
+        if self._one_process_per_chain:
+            return self._args.compose_command(
+                idx,
+                csv_file=self.csv_files[idx],
+                diagnostic_file=self.diagnostic_files[idx]
+                if self._args.save_latent_dynamics
+                else None,
+                profile_file=self.profile_files[idx]
+                if self._args.save_profile
+                else None,
+            )
+        else:
+            return self._args.compose_command(
+                idx,
+                csv_file=self.file_path('.csv'),
+                diagnostic_file=self.file_path(".csv", extra="-diagnostic")
+                if self._args.save_latent_dynamics
+                else None,
+                profile_file=self.file_path(".csv", extra="-profile")
+                if self._args.save_profile
+                else None,
+                num_chains=self._chains,
```

Comment:
ah, I see why `num_chains` is part of the compose for the general args class and not SamplerArgs. I think it's fine for now, barring a bigger refactor of the args classes.